src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/LocationsService.kt

```diff
@@ -21,4 +27,26 @@ class LocationsService(
       ?.let { locationsInsidePrisonClient.getVideoLinkLocationsAtPrison(prisonCode) }
       ?.filter { !enabledOnly || it.active }
       ?.toModel() ?: emptyList()
+
+  @Transactional(readOnly = true)
+  fun getDecoratedVideoLocations(prisonCode: String, enabledOnly: Boolean): List<Location> {
+    val prisonLocations = prisonRepository.findByCode(prisonCode)
+      ?.let { locationsInsidePrisonClient.getVideoLinkLocationsAtPrison(prisonCode) }
+      ?.filter { !enabledOnly || it.active }
+      ?.toModel() ?: emptyList()
+
+    // TODO: Associate by UUID (locationId) rather than the key (MDI-VIDEO-ROOM-1) - the key can change!
+    // But this is a wider change throughout the API when dealing with locations - to follow.
+    val locationsById = prisonLocations.associateBy { it.key }
```

Comment:
A bit puzzled why you are using the key and not the UUID?  It is present on the prison locations ...

Reply:
Changed now. Bit of a pain, as it involved a lot of changes to stubs and integration tests for all locations, which were being set up by locationKey only, but need a predictable UUID to check.